bbb-meeting-api/src/main/scala/org/bigbluebutton/api/join/JoinService.scala

```diff
@@ -0,0 +1,83 @@
+package org.bigbluebutton.api.join
+
+import org.apache.commons.lang3.RandomStringUtils
+import org.bigbluebutton.api.create.CreateParams
+import org.bigbluebutton.common2.api.RegisterUser
+import org.bigbluebutton.service.{ParamsUtils, Utils}
+
+object Roles {
+  val ROLE_MODERATOR = "MODERATOR"
+  val ROLE_ATTENDEE = "VIEWER"
+
+  val LIST = List(ROLE_MODERATOR, ROLE_ATTENDEE)
+}
+
+object JoinService {
+  @throws(classOf[Exception])
+  def createRegisterUser(
+                          meetingId:            String,
+                          params: Map[String, String],
+  ): RegisterUser = {
+    // val disabledFeaturesAsVector: Vector[String] = disabledFeatures.asScala.toVector
+
+    val paramsUtils = ParamsUtils(meetingId, params)
+
+    // We preprend "w_" to our internal meeting Id to indicate that this is a web user.
+    // For users joining using the phone, we will prepend "v_" so it will be easier
+    // to distinguish users who doesn't have a web client. (ralam june 12, 2017)
+    val internalUserID = "w_" + RandomStringUtils.randomAlphanumeric(12).toLowerCase
+    val externUserID = params.getOrElse(CreateParams.ROLE, internalUserID)
+    val authToken = RandomStringUtils.randomAlphanumeric(12).toLowerCase
+
+    // Now determine if this user is a moderator or a viewer.
+    val role : String = {
+      if(!params.getOrElse(CreateParams.ROLE, "").isEmpty && Roles.LIST.contains(params.getOrElse(CreateParams.ROLE, "").toUpperCase)) {
+        params.getOrElse(CreateParams.ROLE, "").toUpperCase
+      } else if(!params.getOrElse(CreateParams.PASSWORD, "").isEmpty) {
+        //TODO not sure if it will be imported since it is DEPRECATED
+        "MODERATOR" //TODO remove it, using for tests
+      } else {
+        ""
+      }
+    }
+    if(role.isBlank) {
+      throw new RuntimeException("Param 'role' is required.")
```

Comment:
Why does a missing role parameter throw an internal server error instead of a response for the user to provide the parameter like it does with fullName, meetingID, and checksum?